### ISSUE TITLE
Menu bug

### ISSUE THREAD

**Issue by vaggelis2018** [thedevdojo/voyager#1342]
- Laravel Version: 5.4.26
- Voyager Version: 0.11.12
- PHP Version: 7.1
- Database Driver & Version:

### Description:
Can anyone help me with this menu display bug? Json_decode expect sting but not an object after creating a new menu with two translations and dynamic route.
### Steps To Reproduce:
I created a new Menu with 2 translations. I added one route item and now i get this error
json_decode() expects parameter 1 to be string, object given
/vendor/tcg/voyager/resources/views/menu/admin.blade.php

**Comment by AdrianKuriata** [comment]
@marktopper @fletch3555 

This is issue what i wrote month ago.

#1101

**Comment by marktopper** [comment]
Maybe this could be a solution: https://github.com/the-control-group/voyager/pull/1101#issuecomment-293860578

**Comment by marktopper** [comment]
@AdrianKuriata, maybe you can test if the comment from above solve the issue? If that is the case, I will reopen your PR so you can make the change if you want.

**Comment by vaggelis2018** [comment]
I can't try it because i delete it from database, created a new one with static url and now i have this problem 
#1343

**Comment by AdrianKuriata** [comment]
@marktopper if it is not a problem i will do this on the weekend, but from my PR i will try your solution and eventually i repair it and will create a new PR.

**Comment by marktopper** [comment]
@AdrianKuriata, sure, no problem 😄

**Comment by vaggelis2018** [comment]
@marktopper i did this
```
protected function prepareParameters($parameters)
    {
        if (is_string($parameters)) {
            $parameters = json_decode($parameters, true);
        } else if (!is_array($parameters)) {
            $parameters = json_decode(json_encode($parameters), true);
        }

        switch (array_get($parameters, 'type')) {
            case 'route':
                $parameters['url'] = null;
                break;
            default:
                $parameters['route'] = null;
                $parameters['parameters'] = '';
                break;
        }

        if (isset($parameters['type'])) {
            unset($parameters['type']);
        }

        return $parameters;
    }
````

and now when i click save i get this error

json_decode() expects parameter 1 to be string, object given (View: /vendor/tcg/voyager/resources/views/menu/admin.blade.php) (View: /vendor/tcg/voyager/resources/views/menu/admin.blade.php)

and still i get this error too #1343

**Comment by AdrianKuriata** [comment]
Yea, i will fix it today or tommorow, you must wait for PR.

**Comment by vaggelis2018** [comment]
ok thank you very much

**Comment by emptynick** [comment]
Fixed by #1355
